src/screens/Home/index.tsx

```diff
@@ -27,7 +35,29 @@ const filterOptions = [
 ];
 
 const Home: React.FC = () => {
+    const {navigate} = useNavigation();
+    const handleNavigateToInternal = ({item}: ResponseProps) => {
+        navigate('Internal', {item});
+    };
     const [selectedOption, setSelectedOption] = useState<number>(1);
+    const [loading, setLoading] = useState(false);
+    const [data, setData] = useState<ResponseItemProps[]>([]);
+    const handleFechtData = useCallback(async () => {
+        try {
+            setLoading(true);
+            const response = await getData();
+            setData(response.results);
+        } catch (error) {
+            console.log(error);
+        } finally {
+            setLoading(false);
+        }
+    }, []);
+
+    useEffect(() => {
+        handleFechtData();
+    }, [handleFechtData]);
+
```

Comment:
Comment on each section to organize the code by its use type

Reply:
resolved in commit https://github.com/IagoWag/rickandmorty/commit/a71461893b6fea906309d9e3507e0c49e0d139f5